### PR TITLE
Some clean up from last PR

### PR DESCRIPTION
**Pull request by BNMetrics**

### CODE REVIEW

**Comment by peterebden** [comment]
FYI the OSX build will be fixed if you rebase off latest master.

File: tools/build_langserver/langserver/analyzer.go [hunk 1]
```diff
@@ -139,32 +132,46 @@ func (a *Analyzer) builtInsRules() error {
...
-			}
+
+			log.Info("Loading built-in build rules...")
+			a.loadBuiltinRules(stmts, string(asset))
 		}
 	}
 
-	a.BuiltIns = statementMap
-	a.Attributes = attrMap
+	for _, i := range a.State.Config.Parse.PreloadBuildDefs {
```
Comment (peterebden) [inline tools/build_langserver/langserver/analyzer.go]: super nit: I would normally use `i` for the first range variable, i.e. the index, and something else for the value (`buildDef` or `bd` or `b` etc...)

File: tools/build_langserver/langserver/analyzer.go [hunk 2]
```diff
@@ -139,32 +132,46 @@ func (a *Analyzer) builtInsRules() error {
...
+		log.Debug("Preloading build defs from %s...", i)
+		a.loadBuiltinRules(stmts, string(bytecontent))
+
+	}
 	return nil
 }
 
+func (a *Analyzer) loadBuiltinRules(stmts []*asp.Statement, fileContent string) {
+	for _, statement := range stmts {
+		if statement.FuncDef != nil && !strings.HasPrefix(statement.FuncDef.Name, "_") {
```
Comment (peterebden) [inline tools/build_langserver/langserver/analyzer.go]: can you not use `statement.FuncDef.IsPrivate` here?

Comment (BNMetrics) [reply tools/build_langserver/langserver/analyzer.go]: Oh yeah! I have that now right? Copy & paste... lol

Comment (BNMetrics) [reply tools/build_langserver/langserver/analyzer.go]: Oh no! the IsPrivate is only for args, I'm gonna add to funcDefs as well

**Pull request merged**